packages/sonarwhal/src/lib/utils/misc.ts

```diff
@@ -40,11 +40,16 @@ const delay = (millisecs: number): Promise<object> => {
     });
 };
 
+/** Try to determin the resource's file name. */
+const getFileName = (resource: string) => {
+    return path.basename(url.parse(resource).pathname);
```

Comment:
Why not use [`path.basename`](https://nodejs.org/dist/latest-v8.x/docs/api/path.html#path_path_basename_path_ext)